src/main/java/com/unstoppabledomains/resolution/Resolution.java

```diff
@@ -78,6 +78,20 @@ public Network getNetwork(NamingServiceType type) {
         return service.getNetwork();
     }
 
+    @Override
+    public String getRecord(String domain, String recordKey) throws NamingServiceException {
+        NamingService service = findService(domain);
+        try {
+            return service.getRecord(domain, recordKey);
+        } catch(NamingServiceException exception) {
+            if (exception.getCode() == NSExceptionCode.RecordNotFound) {
+                throw new NamingServiceException(NSExceptionCode.RecordNotFound, 
+                    new NSExceptionParams("d|c", domain, recordKey));
+            }
+            throw exception;
+        }
```

Comment:
This seems not really needed. we can probably simplify it and let the error pass by. 
```suggestion
        NamingService service = findService(domain);
        return service.getRecord(domain, recordKey);
```